cfgmgr/portmgr.cpp

```diff
@@ -117,14 +81,14 @@ void PortMgr::doTask(Consumer &consumer)
 
         if (op == SET_COMMAND)
         {
-            if (!isPortStateOk(alias))
-            {
-                SWSS_LOG_INFO("Port %s is not ready, pending...", alias.c_str());
-                it++;
-                continue;
-            }
+            /* portOk=true indicates that the port has been created in kernel.
+             * We should not call any ip command if portOk=false. However, it is
+             * valid to put port configuration to APP DB which will trigger port creation in kernel.
+             */
+            bool portOk = isPortStateOk(alias);
```

Comment:
I dont think we need to change this? portsyncd does the `handlePortConfigFromConfigDB` . So portmgrd can still wait and keep the current flow.

Reply:
Hi @prsunny , `handlePortConfigFromConfigDB` is only called once at init stage. It cannot handle case like dynamic port breakout. So, I suppose we need keep this change. What do you think?

Reply:
ok. so the logic here retries for some part of the configuration (kernel), but write to app_db in the first place.

Reply:
yes

Reply:
My concern is, developer has to make sure kernel writes cannot be attempted if portOk is false which is different from the previous logic.  Secondly, it keeps re-writing app_db during the wait period. We need to revisit this logic

Reply:
Thanks for your comment, any suggestion for this?

Reply:
how about doing this in the `if case ` and keep the original logic. 

```
for (auto &entry : field_values)
            {
                writeConfigToAppDb(alias, fvField(entry), fvValue(entry));
```

Reply:
First let me confirm your solution:
```cpp
if (!isPortStateOk(alias))
{
    SWSS_LOG_INFO("Port %s is not ready, pending...", alias.c_str());
    for (auto& i : kfvFieldsValues(t))
    {
        writeConfigToAppDb(alias, fvField(i), fvValue(i));
    }
    it++;
    continue;
}

// rest of the code keeps as original one
```

This solution makes code much clear and simple. But It will repeatedly put the configuration to APP DB until the port state turns "ok".

Reply:
But its the same with your current implementation as well, as you are looping through all retry attributes and writing to app_db, correct?

Reply:
no, for the retry process there are 3 cases:
1. port turns ok, mtu / admin_status will be configured into kernel, other fields will not be put into APP DB again
2. port is still not ok and mtu/admin_status were changed by user, it will only put the new mtu / admin_status to APP DB
3. port is still not ok and mtu / admin_status were not changed, nothing will be put into APP DB